playbooks/roles/streisand-gateway/templates/vhost.j2

```diff
@@ -9,23 +9,43 @@ server {
   ssl_certificate      {{ nginx_self_signed_certificate }};
   ssl_certificate_key  {{ nginx_private_key }};
 {% endif %}
-  add_header Strict-Transport-Security max-age=631138519;
 
   auth_basic            "Authorization Required";
   auth_basic_user_file  {{ streisand_gateway_htpasswd_file }};
 
+  ssl_protocols TLSv1.2;
+  ssl_prefer_server_ciphers on;
   ssl_dhparam {{ dhparams_location }};
+  ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA;
+  ssl_ecdh_curve secp384r1;
+  ssl_session_timeout  10m;
+  ssl_session_cache shared:SSL:10m;
+  ssl_session_tickets off;
+  ssl_stapling on;
+  ssl_stapling_verify on;
```

Comment:
I believe this also requires setting `ssl_trusted_certificate`.

Reply:
I think that when using a LE cert, the fullchain.pem already includes ca-certificate.pem, but if you wan't I can add it either way.

Reply:
The [nginx docs for `ssl_stapling_verify`](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_stapling_verify) specifically say:
> For verification to work, the certificate of the server certificate issuer, the root certificate, and all intermediate certificates should be configured as trusted using the ssl_trusted_certificate directive.

I believe it's not sufficient to have a full chain in the `ssl_certificate` file directive.